core/src/main/java/com/sequenceiq/cloudbreak/service/stack/flow/StackScalingService.java

```diff
@@ -71,6 +83,34 @@
                 userdata.get(InstanceGroupType.CORE), scalingAdjustment, instanceGroupName);
     }
 
+    public void removeInstance(Long stackId, String instanceId) throws Exception {
+        Stack stack = stackService.getById(stackId);
+        InstanceMetaData instanceMetaData = instanceMetaDataRepository.findByInstanceId(stackId, instanceId);
+        String instanceGroupName = instanceMetaData.getInstanceGroup().getGroupName();
+        String hostName = instanceMetaData.getDiscoveryFQDN();
+        eventService.fireCloudbreakInstanceGroupEvent(stack.getId(), UPDATE_IN_PROGRESS.name(),
+                String.format("Start to terminate host '%s' from '%s'.", hostName, instanceGroupName), instanceGroupName);
+        HostMetadata hostMetadata = hostMetadataRepository.findHostsInClusterByName(stack.getCluster().getId(), hostName);
```

Comment:
I think cluster==null should be checked here. If it is null then the instance itself can be removed.